src/GLM.jl

```diff
@@ -1,31 +1,41 @@
 __precompile__()
 
 module GLM
-    using Reexport
-    @reexport using Distributions
+    using Distributions
     using Base.LinAlg.LAPACK: potrf!, potrs!
     using Base.LinAlg.BLAS: gemm!, gemv!
     using Base.LinAlg: QRCompactWY, Cholesky, BlasReal
     using StatsBase: StatsBase, CoefTable, StatisticalModel, RegressionModel
     using StatsFuns: logit, logistic
+    using StatsModels: @formula, Formula, ModelFrame, ModelMatrix
```

Comment:
I wonder whether we should reexport StatsModels for now, to avoid the need to call `using StatsModels` explicitly.

Reply:
I really dislike reexporting entire packages.  Often I will check
```jl
whos(PkgName)
```
to see what the functions and types in a package are and I hate it when everything in, say, `Distributions` is in that list.

Reply:
Yes, I'm not sure what's the best approach here. @ararslan @kleinschmidt Ideas?

Reply:
I don't see the trouble of having `using StatsModels` inside the package; if GLM users want to mess about with statsmodels stuff then I think it's reasonable to have them explicitly ask for it, right?

Reply:
Yeah, my concern is about people who just want to fit a GLM. It would be too bad to require them to run `using StatsModels`.

Reply:
They'd just need `@formula` for that though, right?  I'm not really opposed to re-exporting everything from statsmodels.

Reply:
For the basic use case, yes. But then it sounds natural to allow using contrast types without importing StatsModels explicitly either.

A possibility would be to set up the Stats meta-package we've been talking about for some time, and reexport StatsModels from it. At least it would be simple to tell people who want to do stats to  always call  `using Stats`.

Reply:
I kind of assumed when we were making StatsModels that it would be like StatsBase in that it would typically be extended and reexported. I agree that requiring that users remember to do `using StatsModels` before being able to fit a GLM with a `@formula` seems unfortunate. How did we handle this previously with formulas, which lived in DataFrames?

Reply:
The stats models stuff in dataframes provided a high-level interface to the matrix+vector type fitting methods in GLM.  AFAIK GLM didn't depend on data frames...

Reply:
Ah and as far as I can tell none of the changes to GLM require StatsModels...just for tests.  I think a "batteries" package is probably needed here, something that gives you DataFrames, StatsModels, CSV, maybe Feather.  Previously you'd get the formula methods by `using DataFrames, GLM`.  So from that perspective it's not crazy to have people `using DataFrames, StatsModels, GLM`, or `using Stats, GLM`.

Reply:
> or `using Stats, GLM`.

Though I would think that GLM would be included in said meta-package.

Reply:
I misunderstood the issue @nalimilan was getting at with his original comment.  I hadn't been thinking about setting contrasts, etc.

On looking at this again, I would be fine with replacing this line with `using StatsModels` and exporting all the symbols currently in `StatsModels`.  I prefer not to use the `Reexport` package myself.  I would rather have the default be not to reexport newly added generics and types.  When using `@reexport` you commit yourself to export any new symbols exported by `StatsModels`.

Reply:
I suppose that is StatsModels is totally divorced from DataFrames it may be lightweight enough to have as a dependency; but right now requiring it for GLM essentially also requires DataFrames which I think is not desirable, right?

Reply:
That's a good point. You could just manually reexport any functions you've extended plus any convenience things like `@formula`.

Reply:
> but right now requiring it for GLM essentially also requires DataFrames which I think is not desirable, right?

Another good point. The original plan of having an abstract table interface seems to have fallen by the wayside, so what's the current thinking for having a DataFrames dependency in StatsModels? Seems less than ideal but there are inherently some table-specific things in there.

Reply:
Yeah dealing with tabular data is kind of the whole raison d'etre of StatsModels. I think it should be doable to use something more lightweight (like the column store from DataStreams); the only thing that's needed is column names, types, and something like a view that skips rows with missing values.

Reply:
The situation isn't worse than before at least. It will improve as soon as we create that table interface.

Stats would probably include GLM, but we need to find a general pattern which will also work for modeling packages which would not be in Stats. Honestly, I'd find it a bit ridiculous to tell people that if they do `using MyModeling package`, they also need to do `using StatsModels`. Re-exporting sounds more logical. Ideally at some point there will be a way to load a package automatically when loading another one, without reexporting.

Reply:
The situation may not be worse but this PR changes the relationship between GLM and StatsModels. Before, if you just did `using GLM` you wouldn't get DataFrames or any of the StatsModeling functionality; you'd need to also do `using DataFrames`.  I'm fine with this PR pulling in StatsModels and DataFrames whenever I do `using GLM` now, because I'm never fitting a GLM to non-tabular data anyway.  But there may be people out there who don't want the added dependency on DataFrames and could be surprised by it when they update.  Just something to consider.

Reply:
> But there may be people out there who don't want the added dependency on DataFrames and could be surprised by it when they update. Just something to consider.

Yes, hopefully we'll get rid of the dependency on DataFrames at some point. Until then I'd say it's fine if those people need to install DataFrames.

Reply:
The question was, isn't it weird that you need to call `using StatsModels` to adjust contrasts?

Reply:
In that case I think we should go with your original suggestion of re-exporting the symbols from `StatsModels`, as there aren't very many.  I will amend the PR accordingly.